src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/UserRepositoryTests.cs

```diff
@@ -483,6 +484,107 @@ public async Task UserDoesNotExist_ReturnsNull()
+                Assert.NotNull(result);
+                Assert.Equal(user.Id, result.Id);
+            }
+        }
+
+        [Fact]
+        public async Task UserWithPendingInvite_ReturnsUser()
+        {
+            User user = _dataFixture.DefaultUserWithPendingInvite();
+
+            var contentDbContextId = Guid.NewGuid().ToString();
+
+            await using (var contentDbContext = InMemoryApplicationDbContext(contentDbContextId))
+            {
+                contentDbContext.Users.Add(user);
+                await contentDbContext.SaveChangesAsync();
+            }
+
+            await using (var contentDbContext = InMemoryApplicationDbContext(contentDbContextId))
+            {
+                var repository = BuildRepository(contentDbContext);
+                var result = await repository.FindUserById(user.Id);
+                Assert.NotNull(result);
+                Assert.Equal(user.Id, result.Id);
+            }
+        }
+
+        [Fact]
+        public async Task UserWithExpiredInvite_ReturnsUser()
+        {
+            User user = _dataFixture.DefaultUserWithExpiredInvite();
+
+            var contentDbContextId = Guid.NewGuid().ToString();
+
+            await using (var contentDbContext = InMemoryApplicationDbContext(contentDbContextId))
+            {
+                contentDbContext.Users.Add(user);
+                await contentDbContext.SaveChangesAsync();
+            }
+
+            await using (var contentDbContext = InMemoryApplicationDbContext(contentDbContextId))
+            {
+                var repository = BuildRepository(contentDbContext);
+                var result = await repository.FindUserById(user.Id);
+                Assert.NotNull(result);
+                Assert.Equal(user.Id, result.Id);
+            }
+        }
+
+        [Fact]
+        public async Task SoftDeletedUser_ReturnsNull()
+        {
+            User user = _dataFixture.DefaultSoftDeletedUser();
+
+            var contentDbContextId = Guid.NewGuid().ToString();
+
+            await using (var contentDbContext = InMemoryApplicationDbContext(contentDbContextId))
+            {
+                contentDbContext.Users.Add(user);
+                await contentDbContext.SaveChangesAsync();
+            }
+
+            await using (var contentDbContext = InMemoryApplicationDbContext(contentDbContextId))
+            {
+                var repository = BuildRepository(contentDbContext);
+                var result = await repository.FindUserById(user.Id);
+
+                Assert.Null(result);
+            }
+        }
+
+        [Fact]
+        public async Task UserDoesNotExist_ReturnsNull()
+        {
+            await using var contentDbContext = InMemoryApplicationDbContext();
+
+            var repository = BuildRepository(contentDbContext);
+            var result = await repository.FindUserByEmail("test@test.com");
+            Assert.Null(result);
+        }
```

Comment:
This doesn't look right - it should be calling `FindUserById` rather than `FindUserByEmail`, and using a random ID that's not a saved User.